### PR TITLE
Now the application support any Index/Equity from infomercados

### PR DESCRIPTION
**Pull request by pianista215**
We want to be able to launch different StocksRetrievers with different options (for example, one for Ibex35, other for an equity (BBVA) and other for DOW) without change the code

### CODE REVIEW

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/Main.scala [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-import com.tapiporla.microservices.retrievers.indices.ibex35.Ibex35Coordinator
+import com.tapiporla.microservices.retrievers.indices.stock.StockCoordinator
 
 object Main {
 
   def main(args: Array[String]): Unit = {
 
     val system = ActorSystem("retrievers")
-    val testRetriever = system.actorOf(Props[Ibex35Coordinator])
+    val testRetriever = system.actorOf(Props[StockCoordinator])
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/Main.scala]: I totally discourage using this ```Props``` apply. It implies that the actor you want to create has a class constructor with zero arguments. I prefer to define a ```def apply(): StockCoordinator``` at the companion object for ```StockCoordinator``` and then use ```Props(StockCoordinator())```

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/TapiporlaConfig.scala [hunk 2]
```diff
@@ -31,4 +33,16 @@ object TapiporlaConfig {
     val endpoint: String = scrapyConfig.getString("endpoint")
   }
 
+  object Stats {
+    import scala.collection.JavaConverters._
+    val mmGeneration: Seq[Int] = statsConfig.getIntList("mmGeneration").asScala.map{x => x.intValue()}
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/TapiporlaConfig.scala]: what about
```scala
statsConfig.getIntList("mmGeneration").asScala.map(_.intValue())
```

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 
 /**
   * Chief actor in charge of coordination between Historic Manager and Stat Manager
   */
-class Ibex35Coordinator extends TapiporlaActor {
+class StockCoordinator extends TapiporlaActor {
 
   val historicManager =
-    context.actorOf(Props[Ibex35HistoricManager], name = "Ibex35HistoricManager")
+    context.actorOf(Props[StockHistoricManager], name = s"${stockName}HistoricManager")
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala]: Props apply (mentioned above)

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 
   val statManager =
-    context.actorOf(Props[Ibex35StatManager], name = "Ibex35StatManager")
+    context.actorOf(Props[StockStatManager], name = s"${stockName}StatManager")
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala]: Props apply (mentioned above)

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala [hunk 5]
```diff
@@ -16,39 +16,39 @@ import scala.concurrent.duration._
...
-  * Actor in charge of the Ibex35 data retrieved from ScrapyRT
+  * Actor in charge of the Stock data retrieved from ScrapyRT
   * and persisted in ElasticSearch
   */
-class Ibex35HistoricManager extends Retriever with Stash {
+class StockHistoricManager extends Retriever with Stash {
 
   val esDAO =
-    context.actorOf(Props[Ibex35ESDAO], name = "Ibex35ESDAO_Historic")
+    context.actorOf(Props[StockESDAO], name = s"${stockName}ESDAO_Historic")
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala]: Props apply (mentioned above)

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala [hunk 6]
```diff
@@ -16,39 +16,39 @@ import scala.concurrent.duration._
...
 
   val scrapyDAO =
-    context.actorOf(Props[Ibex35ScrapyDAO], name = "Ibex35ScrapyDAO_Historic")
+    context.actorOf(Props[StockScrapyDAO], name = s"${stockName}ScrapyDAO_Historic")
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala]: Props apply (mentioned above)

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala [hunk 7]
```diff
@@ -79,20 +79,20 @@ class Ibex35HistoricManager extends Retriever with Stash {
...
-        scrapyDAO ! RetrieveIbexDataFrom(date)
+        scrapyDAO ! RetrieveStockDataFrom(date)
       } getOrElse
-        scrapyDAO ! RetrieveAllIbexData
+        scrapyDAO ! RetrieveAllStockData
 
-    case IbexDataRetrieved(historicData) =>
-      log.info(s"Data retrieved from Ibex35 to be updated ${historicData.length}")
+    case StockDataRetrieved(historicData) =>
+      log.info(s"Data retrieved from ${stockName} to be updated ${historicData.length}")
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala]: You can omit braces when mixing simple expressions and string interpolation

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/model/StockHistoric.scala [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+import com.github.nscala_time.time.Imports._
+import com.sksamuel.elastic4s.Hit
+import com.tapiporla.microservices.retrievers.common.TapiporlaConfig
+import com.tapiporla.microservices.retrievers.common.model.{DataInputExtractable, ElasticDocumentInsertable}
+import com.tapiporla.microservices.retrievers.common.stats.StatsGenerator.StatDataInput
+import com.tapiporla.microservices.retrievers.indices.stock.dao.StockESDAO
+
+object StockHistoric {
+  def fromMap(map: Map[String,String]) = {
```
Comment (JSantosP) [inline akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/model/StockHistoric.scala]: Seriously, I would call it ```values``` or whatever XD

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/TapiporlaConfig.scala (hunk 2 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/common/TapiporlaConfig.scala]: :+1: 
Totally agree

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala (hunk 7 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala]: Copy paste is not my best friend xD

Done

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/Main.scala (hunk 1 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/Main.scala]: Sure, I will follow the Akka recoomendation:

object StockCoordinator {
  def props(): Props = Props(new StockCoordinator())
}

And invoke with: StockCoordinator.props()

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala (hunk 3 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala]: Done

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala (hunk 4 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockCoordinator.scala]: Done

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala (hunk 5 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala]: Done

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala (hunk 6 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/StockHistoricManager.scala]: Done

File: akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/model/StockHistoric.scala (hunk 8 above)
Comment (pianista215) [reply akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/stock/model/StockHistoric.scala]: jajajajjajaja XD ok

**Pull request merged**